services/transactionpool/errors.go

```diff
@@ -0,0 +1,14 @@
+package transactionpool
+
+import (
+	"fmt"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+)
+
+type ErrInvalidTransaction struct {
```

Comment:
Today @jlevison and I had a discussion and he says that it's not "dogmatic" to use error types, but rather use `fmt.Errorf`

Reply:
I disagree, look at go's source code, error types are defined like we define them

also, using fmt.Errorf effectively makes all errors strings. Since we want to add extra info to specific errors (like return codes that need to be forwarded to the user), we need rich types and can't rely on strings

Reply:
Let's discuss offline and make a decision

Reply:
I also prefer error types but we need to be consistent with ourselves.